File: Serenity.Script.UI/FilterPanel/FilteringTypes/DecimalFiltering.cs
using jQueryApi;
using Serenity.Data;
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true)]
    public class DecimalFiltering : BaseEditorFiltering
    {
        public DecimalFiltering()
            : base(typeof(DecimalEditor))
        {
        }

        public override List<FilterOperator> GetOperators()
        {
            return AppendNullableOperators(
                AppendComparisonOperators(
                    new List<FilterOperator>()));
        }
    }
}

File: packages/corelib/src/ui/filtering/decimalfiltering.ts
import { nsSerenity } from "../../base";
import { DecimalEditor } from "../editors/decimaleditor";
import { BaseEditorFiltering } from "./baseeditorfiltering";
import { FilterOperator } from "./filteroperator";

/**
 * Filtering handler for decimal fields using a decimal editor.
 */
export class DecimalFiltering extends BaseEditorFiltering<DecimalEditor> {
    static override[Symbol.typeInfo] = this.registerClass(nsSerenity);

    /**
     * Creates a decimal filtering handler.
     */
    constructor() {
        super(DecimalEditor);
    }

    /**
     * Returns the operators supported by this filtering handler.
     * @returns The operators.
     */
    getOperators(): FilterOperator[] {
        return this.appendNullableOperators(
            this.appendComparisonOperators([]));
    }
}
